src/sage/rings/polynomial/laurent_polynomial_ideal.py

```diff
@@ -196,9 +198,27 @@ def __contains__(self, f):
         if not f or f in self.gens():
             return True
         f = self.ring()(f)
-        g = f.__reduce__()[1][0]
+        if self.ring().ngens() > 1:
+            g = f.__reduce__()[1][0]
+        else:
+            g = f.__reduce__()[1][1]
         return (g in self.polynomial_ideal())
 
+    def gens_reduced(self):
+        """
+       A reduced system of generators.
+
+        EXAMPLES::
+
+            sage: P.<x,y> = LaurentPolynomialRing(QQ)
+            sage: J = P.ideal([x^2 - y^-2, x * y^3 + 2 * y^2+ y])
+            sage: J.gens_reduced()
+            (x + 6*y + 5, 3*y^2 + 4*y + 1)
+        """
+        R = self.ring()
+        J = self.polynomial_ideal()
+        return tuple(R(p) for p in J.gens())
```

Comment:
```suggestion
        return tuple([R(p) for p in J.gens()])
```
This is (marginally) faster surprisingly.